test/integration/shoots/maintenance/maintenance.go

```diff
@@ -0,0 +1,442 @@
+		validateFlags()
+
+		shootObject := prepareShoot(f)
+		initialShootForCreation = *shootObject.DeepCopy()
+
+		cloudProfile, err := f.GetCloudProfile(ctx, shootObject.Spec.CloudProfileName)
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		shootMaintenanceTest, err = integrationframework.NewShootMaintenanceTest(ctx, f.GardenClient, cloudProfile, shootObject, shootMachineImageName, f.Logger)
+		gomega.Expect(err).To(gomega.BeNil())
+		testMachineImage.Name = shootMaintenanceTest.ShootMachineImage.Name
+
+		framework.SetupShootWorker(shootMaintenanceTest.Shoot, shootMaintenanceTest.CloudProfile, workerZone)
+		gomega.Expect(err).To(gomega.BeNil())
+		gomega.Expect(len(shootMaintenanceTest.Shoot.Spec.Provider.Workers)).Should(gomega.BeNumerically("==", 1))
+
+		// set machine type & if set, the machineImage name on the first worker image
+		if shootMachineType != nil && len(*shootMachineType) > 0 {
+			shootMaintenanceTest.Shoot.Spec.Provider.Workers[0].Machine.Type = *shootMachineType
+		}
+
+		if shootMachineImageName != nil && len(*shootMachineImageName) > 0 {
+			shootMaintenanceTest.Shoot.Spec.Provider.Workers[0].Machine.Image.Name = *shootMachineImageName
+		}
+
+		if testMachineryRun != nil && *testMachineryRun {
+			f.Logger.Info("Running in test Machinery")
+			// setup the integration test environment by manipulation the Gardener Components (namespace garden) in the garden cluster
+			// scale down the gardener-scheduler to 0 replicas
+			replicas, err := integrationframework.ScaleGardenerScheduler(setupContextTimeout, f.GardenClient.Client(), pointer.Int32Ptr(0))
+			gardenerSchedulerReplicaCount = replicas
+			gomega.Expect(err).To(gomega.BeNil())
+			f.Logger.Info("Environment for test-machinery run is prepared")
+		}
+
+		// the test machine version is being added to
+		prepareCloudProfile(ctx, f)
+		cloudProfileCleanupNeeded = true
+	}, initializationTimeout)
+
+	framework.CAfterEach(func(ctx context.Context) {
+		if shootCleanupNeeded {
+			// Finally we delete the shoot again
+			f.Logger.Infof("Delete shoot %s", shootMaintenanceTest.Shoot.Name)
+			err := f.DeleteShootAndWaitForDeletion(ctx, shootMaintenanceTest.Shoot)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			shootCleanupNeeded = false
+		}
+	}, waitForCreateDeleteTimeout)
+
+	framework.CBeforeEach(func(ctx context.Context) {
+		if shootMaintenanceTest != nil && !shootCleanupNeeded {
+			// set dummy kubernetes version to shoot
+			initialShootForCreation.Spec.Kubernetes.Version = testKubernetesVersionLowMinor.Version
+			// set integration test machineImage to shoot
+			initialShootForCreation.Spec.Provider.Workers[0].Machine.Image = &testMachineImage
+
+			shootMaintenanceTest.Shoot = initialShootForCreation.DeepCopy()
+
+			err := f.GetShoot(ctx, shootMaintenanceTest.Shoot)
+			gomega.Expect(apierrors.IsNotFound(err)).To(gomega.Equal(true))
+			shoot, err := f.CreateShootResource(ctx, shootMaintenanceTest.Shoot)
```

Comment:
I guess you should also wait for the shoots successful creation, or is it not needed?

Reply:
It is not needed as the scheduler is actually scaled down in this serial() test to prevent reconciliation. It only tests the maintenance.